http/src/request/application/update_original_response.rs

```diff
@@ -0,0 +1,391 @@
+    /// if more than 10 embeds are provided.
+    ///
+    /// [the discord docs]: https://discord.com/developers/docs/resources/channel#embed-limits
+    /// [`EMBED_COUNT_LIMIT`]: Self::EMBED_COUNT_LIMIT
+    pub fn embeds(
+        mut self,
+        embeds: Option<Vec<Embed>>,
+    ) -> Result<Self, UpdateOriginalResponseError> {
+        if let Some(embeds_present) = embeds.as_deref() {
+            if embeds_present.len() > Self::EMBED_COUNT_LIMIT {
+                return Err(UpdateOriginalResponseError {
+                    kind: UpdateOriginalResponseErrorType::TooManyEmbeds {
+                        embeds: embeds.expect("embeds are known to be present"),
+                    },
+                    source: None,
+                });
+            }
+
+            for (idx, embed) in embeds_present.iter().enumerate() {
+                if let Err(source) = validate::embed(&embed) {
+                    return Err(UpdateOriginalResponseError {
+                        kind: UpdateOriginalResponseErrorType::EmbedTooLarge {
+                            embeds: embeds.expect("embeds are known to be present"),
+                            index: idx,
+                        },
+                        source: Some(Box::new(source)),
+                    });
+                }
+            }
+        }
+
+        self.fields.embeds.replace(embeds);
+
+        Ok(self)
+    }
+
+    /// Attach a file to the webhook.
+    ///
+    /// This method is repeatable.
+    pub fn file(mut self, name: impl Into<String>, file: impl Into<Vec<u8>>) -> Self {
+        self.files.push((name.into(), file.into()));
+
+        self
+    }
+
+    /// Attach multiple files to the webhook.
+    pub fn files<N: Into<String>, F: Into<Vec<u8>>>(
+        mut self,
+        attachments: impl IntoIterator<Item = (N, F)>,
+    ) -> Self {
+        for (name, file) in attachments {
+            self = self.file(name, file);
+        }
+
+        self
+    }
+
+    /// JSON encoded body of any additional request fields.
+    ///
+    /// If this method is called, all other fields are ignored, except for
+    /// [`file`]. See [Discord Docs/Create Message] and
+    /// [`CreateFollowupMessage::payload_json`].
+    ///
+    /// [`file`]: Self::file
+    /// [`CreateFollowupMessage::payload_json`]: super::CreateFollowupMessage::payload_json
+    /// [Discord Docs/Create Message]: https://discord.com/developers/docs/resources/channel#create-message-params
+    pub fn payload_json(mut self, payload_json: impl Into<Vec<u8>>) -> Self {
+        self.fields.payload_json.replace(payload_json.into());
+
+        self
+    }
+
+    fn request(&mut self) -> Result<Request> {
+        let mut request = Request::builder(Route::UpdateInteractionOriginal {
+            application_id: self.application_id.0,
+            interaction_token: self.token.clone(),
+        });
+
+        if self.files.is_empty() {
```

Comment:
It should not match message create, but `update_webhook_message`

Reply:
as spoken of in discord, there's a bug in `update_webhook_message`, the fix will eventually work its way down here and need to be updated. so that means this also needs updating